src/nimLsp.nim

```diff
@@ -0,0 +1,45 @@
+import std/jsconsole
+import platform/[vscodeApi, languageClientApi]
+
+import platform/js/[jsNodeFs, jsNodePath, jsNodeCp]
+
+from std/strformat import fmt
+from tools/nimBinTools import getNimbleExecPath, getBinPath
+from spec import ExtensionState
+
+proc startLanguageServer(tryInstall: bool, state: ExtensionState) =
+  let rawPath = getBinPath("nimlangserver")
+  if rawPath.isNil or not fs.existsSync(path.resolve(rawPath)):
+    console.log("nimlangserver not found on path")
+    if tryInstall and not state.installPerformed:
+      vscode.window.showInformationMessage("Unable to find nimlangserver, trying to install it via 'nimble install nimlangserver --accept'")
+      state.installPerformed = true
+      discard cp.exec(
+        getNimbleExecPath() & " install nimlangserver --accept",
```

Comment:
On my phone so I can't write the code out as a suggestion could you:

Assign this command build to a let in an earlier line, then use that in the show information message string so they don't go out of sync.

Reply:
sure